test/sequential/rolldown.updateL1FromL1.test.ts

```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
+    await checkAndSwitchMnModeOff();
+  });
+
+  it("[BUG] When a cancel resolution fail, the whole update wont be stored", async () => {
+    await checkAndSwitchMnModeOff();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    const update = new L2Update(api)
+      .withDeposit(
+        txIndex + 1,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_HUNDRED,
+      )
+      .withCancelResolution(txIndex, 1, true)
+      .on(chain)
+      .buildUnsafe();
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update),
+    ).then(async (events) => {
+      expectMGAExtrinsicSuDidSuccess(events);
+    });
+    const event = await waitForEvents(api, "rolldown.RequestProcessedOnL2", 40);
+    const resolutionErr = await getEventError(event);
+    const depositErr = await getEventError(event, 1);
+    expect(depositErr).toEqual(undefined);
+    expect(resolutionErr).toEqual("WrongCancelRequestId");
+    const currencyId = await getAssetIdFromErc20(
+      sequencer.keyRingPair.address,
+      "EthAnvil",
+    );
+    sequencer.addAsset(currencyId);
+    await sequencer.refreshAmounts(AssetWallet.AFTER);
+    expect(sequencer.getAsset(currencyId)?.amountAfter.free!).bnEqual(
+      BN_HUNDRED,
+    );
+    await checkAndSwitchMnModeOff();
+  });
+
+  it("When we have a failed deposit and send it again, it will result in no-execution again", async () => {
+    const update1 = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_TEN_THOUSAND.pow(new BN(18)),
+      )
+      .on(chain)
+      .buildUnsafe();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update1),
+    );
+    const event1 = await waitForEvents(
+      api,
+      "rolldown.RequestProcessedOnL2",
+      40,
+    );
+    const error1 = await getEventError(event1);
+    expect(error1).toEqual("Overflow");
+    const update2 = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_TEN_THOUSAND.pow(new BN(18)),
+      )
+      .withDeposit(
+        txIndex + 1,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_MILLION,
+      )
+      .on(chain)
+      .buildUnsafe();
+    await waitNewBlock();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    const event2 = await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
+    );
+    const eventFiltered = filterAndStringifyFirstEvent(event2, "L1ReadStored");
+    await Rolldown.waitForL2UpdateExecuted(stringToBN(eventFiltered.range.end));
```

Comment:
plz validate that the overflow error event is not in the list of events

Reply:
Done, but I changed a function